### DISCUSSION TITLE
Patched Nanopi neo2 Image

### DISCUSSION THREAD

**Discussion by Tyrrx** [svpcom/wfb-ng#90] [Questions]
Hello,
you said that you have a patched nanopi neo2 kernel, Could you create and upload an image of it because I have difficulties with patching the Kernel for the Neo2?

That would be great :)
David

**Comment by svpcom** [comment]
I don't have clean SD-image with WFB kernel, but build a custom kernel is not very hard. You need to download cross-tools from http://download.friendlyarm.com/nanopineo2
```
mkdir -p /opt/FriendlyARM/toolchain
tar xf gcc-linaro-6.3.1-2017.02-x86_64_aarch64-linux-gnu.tar.xz -C /opt/FriendlyARM/toolchain/
export PATH=/opt/FriendlyARM/toolchain/gcc-linaro-6.3.1-2017.02-x86_64_aarch64-linux-gnu/bin:$PATH
export GCC_COLORS=auto
export ARCH=arm64
export CROSS_COMPILE=aarch64-linux-gnu-
# Then download kernel sources
git clone https://github.com/friendlyarm/linux.git -b sunxi-4.14.y --depth 1
# Apply WFB patch
patch -p1 < wfb.patch
# Apply WFB default config for NEO2
cp .config.wfb .config
make oldconfig
# Build Image, dtb and modules
make Image dtbs modules
# Copy modules to SD card
make modules_install INSTALL_MOD_PATH=/media/SD/rootfs/
```
Then copy Image and dtbs to sd-card
See for details: http://wiki.friendlyarm.com/wiki/index.php/Mainline_U-boot_and_Linux

[wfb.patch.txt](https://github.com/svpcom/wifibroadcast/files/2475712/wfb.patch.txt)
[.config.wfb.txt](https://github.com/svpcom/wifibroadcast/files/2475713/default.config.wfb.txt)

**Comment by svpcom** [comment]
As source SD image you can use ``official-ROMs/nanopi-neo2_sd_friendlycore-xenial_4.14_armhf_YYYYMMDD.img.zip``  from http://download.friendlyarm.com/nanopineo2 or armbian image for NEO2 (if you don't trust neo2 vendor images)

**Comment by Tyrrx** [comment]
what about this? make sunxi_arm64_defconfig ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu-
I tried to patch, which seemed to work fine but after i did an upgrade, i tried to set the txpower to 30 but it still was on 20 max.

**Comment by svpcom** [comment]
Does you card support 30dBm output? Was CRDA patch applied? Run: ``iw reg get``
